File: app/src/main/java/com/mdgd/pokemon/bg/PokemonsLoadingModelFactory.java
package com.mdgd.pokemon.bg;

import com.mdgd.pokemon.models.AppModule;

public class PokemonsLoadingModelFactory {

    public AppModule appComponent;

    public PokemonsLoadingModelFactory(AppModule appComponent) {
        this.appComponent = appComponent;
    }

    public LoadPokemonsContract.ServiceModel create() {
        return new LoadPokemonsModel(appComponent.getPokemonsRepo(), appComponent.getCache());
    }
}

File: app/src/main/java/com/mdgd/pokemon/bg/PokemonsLoadingModelFactory.kt
package com.mdgd.pokemon.bg

import com.mdgd.pokemon.models.AppModule

class PokemonsLoadingModelFactory(var appComponent: AppModule) {

    fun create(): ServiceModel {
        return LoadPokemonsModel(appComponent.getPokemonsRepo(), appComponent.getCache())
    }
}
